packages/fetch-router/demos/node/app/auth.ts

```diff
@@ -0,0 +1,93 @@
+import type { BuildRouteHandler, Middleware, RouteHandlers } from '@remix-run/fetch-router'
+import { html } from '@remix-run/html-template'
+import * as res from '@remix-run/fetch-router/response-helpers'
+
+import { routes } from '../routes.ts'
+import * as templates from './templates.ts'
+import { loginUrl } from './utils.ts'
+
+function requireGuest(): Middleware {
+  return async (context, next) => {
+    let username = context.session.get('username')
+    if (username) {
+      return res.redirect(routes.posts.index.href())
+    }
+    return next()
+  }
+}
+
+export let login = {
+  middleware: [requireGuest()],
+  handlers: {
+    index({ session, url }) {
+      let error = session.get('error') as string | undefined
+      let currentUser = session.get('username') as string | undefined
+      let redirectTo = url.searchParams.get('redirectTo')
+
+      // TODO: check if this should be cleared on subsequent requests, given that we used flash to set the error
+      if (error) {
+        session.unset('error')
+      }
```

Comment:
You can see this in the video, but I was trying to use `session.flash` to temporarily show an error message. It did not behave the way I anticipated, which I'm not sure if that's user error or what, so I had to add this `sesssion.unset('error')` here, which I don't love